### PR TITLE
Updates the image's container size when we resize the window

### LINKED ISSUE #6191: Image resize tool sets dimensions when it shouldn't
As discovered in https://github.com/WordPress/gutenberg/pull/5812#issuecomment-381503328 and possibly a regression since https://github.com/WordPress/gutenberg/pull/5973, it appears the library we use to add resize handles to the image blocks needs a little love.

The issue is that the inline styles that a resized image receives here: https://github.com/WordPress/gutenberg/blob/master/blocks/library/image/block.js#L253 — they get applied when they shouldn't be. That means if you resize the viewport, the image doesn't resize with it, as the dimensions are explicitly set as inline styles.

Steps to reproduce:

1. Set the editor canvas to show at a >782px breakpoint (this is where the navigation bar on the left shows up)
2. Insert an image that's larger than the editor canvas. 
3. Resize the viewport and note that the image doesn't size with it, as it should

That's the buggy part. 

It works, if your editor canvas is <782 breakpoint. I.e. if you repeat the steps above at the breakpoint where the left sidebar is hidden, then no inline styles are set, and the image behaves as it should.

Screenshots:

screen shot 2018-04-16 at 09 44 03

☝️That's at <782px, note no inline styles.

screen shot 2018-04-16 at 09 44 25

☝️That's at >782px, note the inline style.

GIF:

inlinestyles

CC: @youknowriad — I think you worked on image resizing, any idea what this might be a regression from, or how to best fix?

### PR DESCRIPTION
**Pull request by youknowriad**
closes #6191

This PR fixes the resizing behavior of image blocks when we resize the browser size.

**Testing instructions**

 - Repeat the instructions in #6191 
 - The image should scale according to its container size.

### CODE REVIEW

**Comment by jasmussen** [comment]
Not sure if I've checked this branch out correctly, but it doesn't seem to be working for me.

resize

While it sounds like setting the image size on resize, I'm not sure that would fix #6191. The ideal behavior is actually that the dimensions on an un-resized image _aren't set at all_, not that they are re-calculated per the viewport width. The really weird thing is that this works as expected <782px, and the dimensions seem to get added >782px.

Make sense?

**Comment by youknowriad** [comment]
Ok, so this is happening because of the "fit-content width" of the image wrapper. Before we were checking the wrapper's size, but since the wrapper's size don't change with this width, it's creating issues.

Not an easy fix, still looking. I think there's a reason for us setting the size of the image even if we want it to take the parent's size (related to the resizing behavior) I'll see if removing them fixes the issues.

**Comment by youknowriad** [comment]
Removed the explicit width/height. I can't see regressions but please help testing the resizing behavior :)

**Comment by jasmussen** [comment]
Nice, this seems to work perfectly with no regressions! 

The inline styles are still technically applied to the resizablebox div _only when the viewport is >782px_, though, and I don't understand why they instantiate there but not earlier. But if we can't figure out why those are applied, the fix you just pushed seems solid enough, and appears to solve the issue!

Do you know, perchance, why those styles get applied at that breakpoint and not before? If they are there for a reason, then this is definitely good to ship. If not, then I'm a little reluctant that there's another regression that this is a bandaid to.

Thanks for your work!

**Comment by jasmussen** [comment]
> I guess because we disable the resizable box entirely for small viewports.

Awesome, that sounds right, then we know what's up. I think this is good to ship, experience wise, and it solves a real problem in master right now. Thanks so much for this, and CC: @noisysocks, as it fixes the issue I mentioned in your branch.

File: blocks/library/image/block.js [hunk 1]
```diff
@@ -252,8 +252,8 @@ class ImageBlock extends Component {
 						return (
 							<ResizableBox
 								size={ {
-									width: currentWidth,
-									height: currentHeight,
+									width,
+									height,
 								} }
```
Comment (noisysocks) [inline blocks/library/image/block.js]: It probably doesn't matter in practice but [the docs](https://github.com/bokuweb/re-resizable#size--width-number--string-height-number--string-) say that `size` is optional and that `.width` and `.height` are required. 

```js
<ResizableBox
    size={ width && height && {
        width,
        height,
    } }
    ...
/>
```

**Review by noisysocks — APPROVED** [review]
Makes sense and fixes the issue!

File: blocks/library/image/image-size.js [hunk 2]
```diff
@@ -79,4 +80,6 @@ class ImageSize extends Component {
 	}
 }
 
-export default ImageSize;
+export default withGlobalEvents( {
```
[the remark this answers is not in the document]
Comment (youknowriad) [reply blocks/library/image/image-size.js]: cc @aduth

**Comment by jasmussen** [comment]
YOU ARE THE AWESOMEST

**Pull request merged**